src/fiobank/models.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import annotations
+
+import re
+from datetime import date
+from decimal import Decimal
+from typing import Any, Optional, Union, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from datetime import date as DateType
+else:
+    DateType = date
+
+try:
+    from pydantic import BaseModel, Field, field_validator
+    PYDANTIC_AVAILABLE = True
+except ImportError:
+    PYDANTIC_AVAILABLE = False
+    PYDANTIC_AVAILABLE = False
+    
+    # Create minimal mock classes for fallback when pydantic is not available
+    class BaseModel:
+        model_config = {}
+        def __init__(self, **kwargs):
+            for k, v in kwargs.items():
+                setattr(self, k, v)
+        
+        @classmethod
+        def model_validate(cls, data, context=None):
```

Comment:
what is this abomination? this is overly complex and overengineered. omg. I wanted the code be simpler and easier to read, this what you created is like you invented your own astronomical clock just to make a pydantic model